app/controllers/users_controller.rb

```diff
@@ -1,30 +1,73 @@
 class UsersController < ApplicationController
-  def new
-    @user = User.new
-  end
+  before_action :logged_in_user, except: %i(new create show)
+  before_action :load_users, only: [:show, :edit, :correct_user]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
 
-  def show
-    @user = User.find_by id: params[:id]
-    return if @user
-
-    flash[:alert] = t "users.user_notexist"
-    redirect_to root_path
+  def index
+    @users = User.paginate(page: params[:page], per_page: Settings.pages)
   end
 
   def create
     @user = User.new user_params
     if @user.save
       flash[:success] = t ".create_success"
+      log_in @user
       redirect_to @user
     else
       flash[:success] = t ".create_fail"
       render :new
     end
   end
 
+  def new
+    @user = User.new
+  end
+
+  def show; end
+
+  def update
+    if @user.update user_params
+      flash[:success] = t "users.profile_updated"
+      redirect_to @user
+    else
+      flash[:alert] = t "users.profile_updated_fail"
+      render :edit
+    end
+  end
+
+  def edit; end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = "User deleted"
+    redirect_to users_url
+  end
   private
+  def admin_user
+    redirect_to(root_url) unless current_user.admin?
+  end
+
+  def load_users
```

Comment:
```suggestion
  def load_user
```
load 1 user thôi mà để số nhiều chi e